blog_post.md

```diff
@@ -0,0 +1,331 @@
+
+To generate evaluation datasets at scale, we leverage a script [generate_eval_dataset.py](https://github.com/marcgs/lob-chatbot-sample/blob/main/evaluation/chatbot/ground-truth/generate_eval_dataset.py) that combines scenario templates with real or representative business data. This script automates the creation of test cases by filling placeholders in templates with data from support tickets and action items. Below is a simplified example of a scenario template including placeholders that will be replaced with business data:
+
+```json
+{
+    "scenario_name": "create_ticket_and_action_item",
+    "user_instructions": "Create a new support ticket by manually providing the data and create a new action item for it.",
+    "task_completion": "The user has confirmed the end of the session.",
+    "expected_function_calls": [
+      {
+        "functionName": "TicketManagementPlugin-create_support_ticket",
+        "arguments": {
+          "title": "{ticket.title}",
+          "department_code": "{ticket.department_code}",
+          "priority": "{ticket.priority}",
+          "workflow_type": "{ticket.workflow_type}",
+          "description": "{ticket.description}",
+          "expected_outcome": "{ticket.expected_outcome}",
+          "customer_visible": "{ticket.customer_visible}"
+        }
+      },
+      {
+        "functionName": "ActionItemPlugin-create_action_item",
+        "arguments": {
+          "title": "{action.title}",
+          "assignee": "{action.assignee}",
+          "due_date": "{action.due_date}"
+        }
+      }
+    ]
+}
+```
+
+### Simulation Outputs
+
+Throughout the simulated conversation, the framework automatically captures all exchanged messages and every function call made by the agent, including the function name and all arguments provided. This structured record of function calls is essential for evaluation: it allows us to directly compare the chatbot's actions against the expected function calls for each scenario, measuring not just whether the right functions were called, but also whether the correct parameters were supplied and the business process was followed as intended.
+
+Simulation outputs:
+
+![Simulation Outputs](./sim_outputs.png)
+
+Simulated conversation history:
+
+![Chat History](./chat_history.png)
+
+Actual vs expected function calls - used to calculate various metrics and overall reliability score of the LOB agent:
+
+![Function Calls](./function_calls.png)
+
+### Evaluation, Metrics, and Error Analysis
+
+With conversation history, actual and expected function call data in hand, we can use the framework to compute a suite of evaluation metrics, which include:
+
+- **Function Call Name Precision and Recall**  
```

Comment:
Should we reorganize the metrics by:
- Primary metrics:
  - Function Call Name Recall
  - Function Call Parameter Recall
 - Auxiliary metrics:
   - Function Call Name Precision
   - Function Call Parameter Precision
   - Reliability Score

Reply:
I think we don't need to do this. Current structure makes more from metrics perspective and in Reliability Score we explain recall is our primary metric